types/signed_header.go

```diff
@@ -103,31 +116,37 @@ func (sh *SignedHeader) ValidateBasic() error {
 		return err
 	}
 
-	// Handle Based Rollup case
-	if sh.Validators == nil || len(sh.Validators.Validators) == 0 {
-		return nil
-	}
-
 	if err := sh.Validators.ValidateBasic(); err != nil {
 		return err
 	}
 
+	// Rollkit vA enforces a centralized sequencer.
```

Comment:
consider moving newly added checks to a separate function like:
```
func (sh *SignedHeader) checkCentralizedSequencer() error {
	validators := sh.ValidatorSet.Validators
	if len(validators) != 1 {
		return errors.New("cannot have more than 1 validator (the centralized sequencer)")
	}
	// make sure the proposer address is the same as the first validator in the validator set
	first := validators[0]
	if !bytes.Equal(first.Address.Bytes(), sh.ProposerAddress) {
		return errors.New("proposer address in SignedHeader does not match the expected centralized sequencer address")
	}
	// check proposer against the first validator in the validator set
	proposer := sh.Proposer
	if proposer.PubKey.Equals(first.PubKey) &&
		bytes.Equal(proposer.Address.Bytes(), first.Address.Bytes()) &&
		proposer.VotingPower == first.VotingPower &&
		proposer.ProposerPriority == first.ProposerPriority {
		return errors.New("proposer and first validator in the validator set are not same")
	}
}
```
to keep the main validation function clean and easier to read.

Reply:
should i put the signature check in here as well?

Reply:
no need. signature check can be outside as before.